objects/src/accounts/package/storage_entry/template.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use vm_core::{Felt, Word};
+use vm_processor::Digest;
+
+use crate::accounts::package::AccountComponentTemplateError;
+
+// TEMPLATE KEY
+// ================================================================================================
+
+// A simple wrapper type around a string key, used to enable templating.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TemplateKey {
+    key: String,
+}
+
+impl TemplateKey {
+    pub fn new(key: impl Into<String>) -> Self {
+        Self { key: key.into() }
+    }
+
+    pub fn inner(&self) -> &str {
+        &self.key
+    }
+}
+
+impl TryFrom<&str> for TemplateKey {
```

Comment:
I'm wondering if `FromStr` would be more appropriate, so one can call `"{{my.key}}".parse()?`. I'm never quite sure though when to use `TryFrom<&str>` versus `FromStr` so feel free to disregard.

Reply:
I actually debated this as well but it didn't seem like we used `FromStr` too much across projects, so I went with this instead.